### PR TITLE
Update <ShippingRates />

### PR DESCRIPTION
**Pull request by v18**
This PR:
- removed unnecessary props from `<ShippingRates />`
- removed boolean parameter from `renderRateNotice()`
- change `renderTitle` to `getTitle`

### CODE REVIEW

File: client/shipping-label/views/purchase/steps/rates/list.js [hunk 1]
```diff
@@ -63,7 +61,7 @@ const ShippingRates = ( {
 
 	return (
 		<div>
-			{ renderRateNotice( showRateNotice ) }
+			{ shouldShowRateNotice && renderRateNotice() }
```
Comment (nabsul) [inline client/shipping-label/views/purchase/steps/rates/list.js]: Are you sure this is equivalent to the old code? How does the value `false` get rendered in this div?

Comment (nabsul) [inline client/shipping-label/views/purchase/steps/rates/list.js]: Just to make the code clearer, you could go with something like:
`shouldShowRateNotice ? renderRateNotice() : ''`

PS: the previous implementation of `renderRateNotice()` returning a null wasn't much better. It should have also returned an empty string if the boolean was false.

Comment (v18) [inline client/shipping-label/views/purchase/steps/rates/list.js]: React will ignore non-0 falsey values, so if `shouldShowRateNotice` evaluates to `false` or `null`, it won't get rendered by React ([ref](https://facebook.github.io/react/docs/jsx-in-depth.html#booleans-null-and-undefined-are-ignored)). There isn't a need to return an empty string in the false case, then.

The `&&` is a common pattern for condition rendering ([ref](https://facebook.github.io/react/docs/conditional-rendering.html#inline-if-with-logical--operator)), and we also use it frequently in Calypso, so would prefer to keep that.

**Comment by nabsul** [comment]
Looks good. :shipit:

**Pull request merged**